internal/gitinterface/changes.go

```diff
@@ -119,3 +160,23 @@ func diff(treeA, treeB *object.Tree) ([]string, error) {
 
 	return paths, nil
 }
+
+type diffHeap []string
```

Comment:
This isn't something we want to do in this PR but the other funcs here could probably also use this. We can discuss this separately if you want to work on that as well.

Reply:
I would be open to creating a PR to make this into a general function for use in other parts of the project.

Reply:
Would a generalized heap that could take in any type, and a comparator function, be more useful than this heap that only accepts strings and is a min heap, since it would be more generalized and could be used anywhere in the project for any type of datastructure?

Reply:
Perhaps an analysis of where / why / what we sort in gittuf is in order first? :smile: